compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -1993,6 +1993,9 @@ public void visit(BLangRecordTypeNode recordTypeNode) {
         for (BLangSimpleVariable field : recordTypeNode.fields) {
             addTypeDependency(tsymbol, Types.getReferredType(field.getBType()), new HashSet<>());
             analyzeNode(field, env);
+            for (BLangAnnotationAttachment annotationAttachment : field.annAttachments) {
+                analyzeNode(annotationAttachment.expr, env);
+            }
```

Comment:
Was this change to address a specific bug? Have we added tests? Seems like we aren't doing it for tuples either.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/38487
Fix for tuple annotation and added test cases